rotkehlchen/globaldb/handler.py

```diff
@@ -1203,6 +1225,86 @@ def get_historical_price_data(source: HistoricalPriceOracle) -> List[Dict[str, A
              'to_timestamp': entry[3],
              } for entry in query]
 
+    @staticmethod
+    def rebuild_assets_list(user_db_conn: sqlite3.Connection) -> Tuple[bool, str]:
+        """
+        Delete all custom asset entries and repopulate from the last
+        builtin version
+        """
+        root_dir = Path(__file__).resolve().parent.parent
+        builtin_database = root_dir / 'data' / 'global.db'
+        detach_database = 'DETACH DATABASE "clean_db";'
+
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        user_db_cursor = user_db_conn.cursor()
+
+        try:
+            # First chech that the operation can be made. For this create a set with the
+            # identifiers on the user db and the db packaged and make a difference. If the
+            # difference is not the empty set the operation is dangerous and the user should
+            # be notified.
+            query = cursor.execute('SELECT identifier from assets;')
+            user_ids = set(query.fetchall())
+            # Atach the clean db packaged with rotki
+            cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
+            # Get built in identifiers
+            query = cursor.execute('SELECT identifier from clean_db.assets;')
+            shipped_ids = set(query.fetchall())
+            diff_ids = user_ids - shipped_ids
+            if len(diff_ids) != 0:
+                # If we have non common elements we have to see if this assets are used anywhere
+                # in the database tables
+                diffs = ', '.join([f'"{id[0]}"' for id in diff_ids])
+                for table, column in TABLES_WITH_ASSETS:
+                    diffs_query = f'SELECT COUNT(*) FROM {table} where {column} in ({diffs})'
+                    user_db_cursor.execute(diffs_query)
+                    amount = user_db_cursor.fetchone()
+                    if amount[0] != 0:
+                        cursor.execute(detach_database)
+                        log.error(
+                            f'Assets inserted by the user are used in the database '
+                            f'tables {diff_ids}',
+                        )
+                        return False, 'There are assets that can not be deleted'
```

Comment:
I think you should let the user know which assets these are here.

Reply:
We save it in the logs. Moving it to the UI could be not that useful I believe.

Reply:
Logs are not always on. The least you would need to do is tell them to check the logs.

Which in the case of not yet activated debug logs would mean they would have to also activate them.